### ISSUE TITLE
Create API automated test scripts to improve reliability

### ISSUE THREAD

**Issue by timstackblock** [hirosystems/stacks-blockchain-api#1493]
**Is your feature request related to a problem? Please describe.**
This feature is to create API automated test scripts to improve reliability. The first script was created Dec 12th.

**Describe the solution you'd like**
- [x] #1542
- [x] #1548
- [x] #1545
- [x] #1546
- [x] #1544
- [x] #1556
- [x] #1558

**Describe alternatives you've considered**
Functional testing is the assessment of specific functions within the codebase. It makes sure the API actually works within the expected parameters, by doing the following:

Returns output for a given input; and handles errors when the results are outside of the expected parameters.
One of the functional testing types is Positive / Negative testing. Negative testing checks how an API responds to wrong input, positive testing verifies the correct functioning of the API when the input is the expected result.

**Additional context**
This is the first pass at this we have currently created one script and will continue to build out the test suite.

**Comment by timstackblock** [comment]
@zone117x @rafaelcr when you get a chance please clone this repo and take a look at the automated test I created https://github.com/timstackblock/apiTest. Once I get your feed back I will continue creating test for the other scenarios.

**Comment by timstackblock** [comment]
Moved this to a Hirosystems repo https://github.com/hirosystems/apiTest

**Comment by timstackblock** [comment]
A new PR was added with several new test https://github.com/hirosystems/apiTest/pull/1 . 

Below I have added a few ideas that I think can help add confidence and quality to API releases. We can talk through these thoughts and see what works well for the API team and what does not then finalize them once I get your feed back. Thank you for your time.

### **TENTATIVE NEXT STEPS**

**1. Get the current test into CI with Github actions.**

_Currently working on adding CI with Github actions to this repo, I am hoping to close this out next week if I run into issues doing so I will try to pull in @CharlieC3 and devops when they have available cycles to assist in this task. My goal is to get the test running with a cron job on the testnet._ 


**2. Outline the work flow of how developers will incorporate the automation test in the current SDLC.**

_Once the test are running on the testnet, developer will be able to deploy the new API code to the testnet and observe the test results over time. This will give them confidence that the changes are working as expected. Ideally after a day or so on the testnet without failure the PR will be ready to move to production._ 

**3. Work with the API team to identify the major regressions they have found in the API and build test to identify them.**

_Automation test can only look for bugs and failures that it was designed to test for. This means we must take some time to identify all of the regressions that have impacted the API team in the past. This product knowledge of past API bugs and the PRs created to fix them can come from the API end users and API team developers. We want to identify the following:_ 

-  _What are the most common regressions that sneak into the code base time and time again?_
-  _What are some common bugs that you run into in production?_
-  _What complaints do we get from developers and users of the API?_

_If we can identify these problems, list them out and build test to look for them, we will increase confidence in our builds when they are released._

**4. Better integrate QA into the API teams current development process**

_With the additions of a full suite of automation test which we are currently working towards we will get test coverage to protect us from regressions. But there are better ways to add quality to the API besides test. Integrating a QA engineer into the current development workflow is an easy way to add quality and another set of eyes on changes going out. In the past we would notify the QA team of a new API release and have them test it for a few days before it was released. Although we are moving away from all manual test I think incorporating a few check manually is a good way to incorporate QA into the sprint with dev.  We can take this a bit further and incorporate a summary of changes made as well as a list of issues that are resolved by the change to give added context on the code change so QA can create adequate acceptance criteria._ 

_Ways we can implement this:_

_When development is working on a new PR, QA is notified of the release date we are working for and when the PR will be ready for test as well as what changes are in the PR. QA will begin to think through how they will test the changes and begin to prepare to add automation test (if necessary) for the new functionality as well as manually look at the error to gain context. When the build is ready for test QA can test the build then watch the build when it is pushed to the testnet and look for test failures. Once this is complete they will notify the API team that they are test passed._

**Comment by timstackblock** [comment]
> 1. Get the current test into CI with Github actions.

> Currently working on adding CI with Github actions to this repo, I am hoping to close this out next week if I run into issues doing so I will try to pull in @CharlieC3 and devops when they have available cycles to assist in this task. My goal is to get the test running with a cron job on the testnet.


This has been completed in https://github.com/hirosystems/apiTest/pull/6

**Comment by timstackblock** [comment]
These 2 are in progress

https://github.com/hirosystems/stacks-blockchain-api/issues/1548
https://github.com/hirosystems/stacks-blockchain-api/issues/1545

**Comment by timstackblock** [comment]
This PR adds Jest assertions and closes issue https://github.com/hirosystems/stacks-blockchain-api/issues/1545

https://github.com/hirosystems/apiTest/pull/9/

**Comment by timstackblock** [comment]
This issue can be closed as the automation test are now being tested against prod data on https://api.dev.hiro.so/ and the testnet scripts are being deprecated
